### ISSUE TITLE
Errors during migration

### ISSUE THREAD

**Issue by matrix07012** [misskey-dev/misskey#4808]
## 💡 Summary

Migration spits out several different errors then gets stuck after migrating emojis and printing "DONE :)".
If I quit it after the "DONE :)" message and start Misskey, several things are broken or didn't migrate.

## 🙂 Expected Behavior

Migration

## ☹️ Actual Behavior

Migration spits out several different errors then gets stuck after migrating emojis and printing "DONE :)".
Here's my entire console output from the migration:
[migrate.txt](https://github.com/syuilo/misskey/files/3118839/migrate.txt)

If I quit it after the "DONE :)" message and start Misskey, thumbnails are gone, drive folders and files withing them are gone and instance settings are gone.

## 📝 Steps to Reproduce

1.Switch to migration branch
2.Edit config
3.`npm i`
4.`npm run build`
5.Create PostgreSQL user and database with:
CREATE USER misskey WITH PASSWORD 'password';
CREATE DATABASE misskey OWNER misskey TEMPLATE template0;
update misskey set encoding = pg_char_to_encoding('UTF8') where datname = 'misskey';
6.`npm run init`
7.`npm run migrate`
8.Force quit after "DONE :)"
9.Switch master
10.'npm install'
11.'NODE_ENV=production npm run build'
12.start Misskey

## 📌 Environment
Ubuntu 18.04.2 LTS (GNU/Linux 4.15.0-46-generic x86_64)
NodeJS v11.11
PostgreSQL 10
MongoDB 4.0.9
Misskey before migration 10.100.0
Migration Commit 747bf3b
Misskey after migration 11.4.0

**Comment by rinsuki** [comment]
@matrix07012 can you try with https://github.com/syuilo/misskey/commit/6ce986fcbf50dfa414608c2f37f7b930f2891d8b commit?

**Comment by matrix07012** [comment]
> @matrix07012 can you try with [6ce986f](https://github.com/syuilo/misskey/commit/6ce986fcbf50dfa414608c2f37f7b930f2891d8b) commit?

Ok. I have to go to sleep, so I will try tomorrow.

**Comment by rinsuki** [comment]
I think, after resolved #4809 , this migration will successful.

**Comment by matrix07012** [comment]
I also forgot to mention that the migration branch doesn't include node modules monk, mongodb and @types/mongodb and you have to manually install them.

@rinsuki
I tried it with commit 318ae00, the error with date disappears, but all the other errors remain.
[migrate.txt](https://github.com/syuilo/misskey/files/3121122/migrate.txt)

**Comment by matrix07012** [comment]
Great job @sftblw . I migrated with only few errors. Drive folders and files in them work. The only thing missing are thumbnails. Is it possible to recreate them after the migration?

Here's my entire console output:
[migrate.txt](https://github.com/syuilo/misskey/files/3170289/migrate.txt)

**Comment by u1-liquid** [comment]
I think you should adjust your database table to make #4809 actually applied.
There was no migration script provided for to do this, but you can still make it from your psql command line.

```sql
ALTER TABLE "user_profile" ALTER COLUMN "description" TYPE VARCHAR(2048);
```
Maybe you can try 4096 if 2048 doesn't fit your current source.

regarding other errors, as rough view, it seems you can ignore them because most of those errors are casued by remote instance note/toots.

**Comment by matrix07012** [comment]
> I think you should adjust your database table to make #4809 actually applied.
> There was no migration script provided for to do this, but you can still make it from your psql command line.
> 
> ```sql
> ALTER TABLE "user_profile" ALTER COLUMN "description" TYPE VARCHAR(2048);
> ```
> 
> Maybe you can try 4096 if 2048 doesn't fit your current source.
> 
> regarding other errors, as rough view, it seems you can ignore them because most of those errors are casued by remote instance note/toots.

Yeah, I know. Those errors cause no actual problems. Bigger issue is the missing thumbnails/file previews.

**Comment by sftblw** [comment]
As a information, I had same problem (missing thumbnails) but fixed by #4903 especially by 076b21a , by adding thumbnail URL to migration. My instance stored files on Amazon S3.

If you did not migrate yet I think checking the thumbnail URL via SQL might be useful.

**Comment by matrix07012** [comment]
> As a information, I had same problem (missing thumbnails) but fixed by #4903 especially by [076b21a](https://github.com/syuilo/misskey/commit/076b21aa87e55905f990400c51ec7abd908dba97) , by adding thumbnail URL to migration. My instance stored files on Amazon S3.
> 
> If you did not migrate yet I think checking the thumbnail URL via SQL might be useful.

I store everything locally.
